README.md

```diff
@@ -133,6 +193,16 @@ use
 
 `cacheFetch(fetch, strategy, cache)`
 
+under the hood:
+
+`cacheFetch` calls `fetch` to return a fresh value only if the value is not present in `cache` or `strategy` returns `false` (value is expired).
+
+Whenever `fetch` is called:
+- `CacheValue` in `cache` is replaced with a new one with `done: undefined` and `blocked: on-going-promise`
+
+Whenever `fetch` completes:
+- `CacheValue` in `cache` is replaced with a new one with `blocked: undefined` and `done: { value: final-value, timestamp: current-timestamp, promise: resolved-promise }`
+
 **catalog**
 
 `catalog: S ~> List[P]`
```

Comment:
@giogonzo here we heavily use the letter `S` instead of `A` (see `ptoa` where `A` is the output and `S` is the input)